docs/site/Loopback-component-authentication.md

```diff
@@ -0,0 +1,603 @@
+   */
+  name: string;
+
+  /**
+   * The 'authenticate' method takes in a given request and returns a user profile
+   * which is an instance of 'UserProfile'.
+   * (A user profile is a minimal subset of a user object)
+   * If the user credentials are valid, this method should return a 'UserProfile' instance.
+   * If the user credentials are invalid, this method should throw an error
+   * If the user credentials are missing, this method should throw an error, or return 'undefined'
+   * and let the authentication action deal with it.
+   *
+   * @param request
+   */
+  authenticate(request: Request): Promise<UserProfile | undefined>;
+}
+```
+
+Developers that wish to create a custom authentication strategy must implement
+this interface. The custom authentication strategy must have a **unique** `name`
+and have an `authenticate` function which takes in a request and returns the
+user profile of an authenticated user.
+
+Here is an example of a basic authentication strategy named
+`BasicAuthenticationStrategy` in `basic-strategy.ts`:
+
+```ts
+export interface Credentials {
+  username: string;
+  password: string;
+}
+
+export class BasicAuthenticationStrategy implements AuthenticationStrategy {
+  name: string = 'basic';
+
+  constructor(
+    @inject(UserServiceBindings.USER_SERVICE)
+    private userService: UserService,
+  ) {}
+
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    const credentials: Credentials = this.extractCredentials(request);
+    const user = await this.userService.verifyCredentials(credentials);
+    const userProfile = this.userService.convertToUserProfile(user);
+
+    return userProfile;
+  }
+
+  extractCredentials(request: Request): Credentials {
+    let creds: Credentials;
+
+    /**
+     * Code to extract the 'basic' user credentials from the Authorization header
+     */
+
+    return creds;
+  }
+}
+```
+
+As you can see in the example, an authentication strategy can inject custom
+services to help it accomplish certain tasks. See the complete examples for
+[basic authentication strategy](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/__tests__/fixtures/strategies/basic-strategy.ts)
+and
+[jwt authentication strategy](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/__tests__/fixtures/strategies/jwt-strategy.ts).
+
+The `AuthenticationComponent` component also provides two **optional** service
+interfaces which may be of use to your application :
+[UserService](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/services/user.service.ts)
+and
+[TokenService](https://github.com/strongloop/loopback-next/blob/master/packages/authentication/src/services/token.service.ts).
+
+Developers that wish to use [passport](http://www.passportjs.org/)-based
+authentication strategies, please see [Passport Authentication Strategies](TBD).
```

Comment:
Is this link going to be added as part of this PR?

Reply:
@nabdelgadir ,

> Is this link going to be added as part of this PR? 

I don't yet know the url or file name that @jannyHou is going to have for her PR... which will cover the passport stuff.  @jannyHou , can you offer a best-guess to your new url?

Reply:
If the link doesn't exist, is it better not to mention it until @jannyHou added her docs?

Reply:
Removing (for now):

```
Developers that wish to use [passport](http://www.passportjs.org/)-based
authentication strategies, please see [Passport Authentication Strategies](TBD).
```